### DISCUSSION TITLE
Problem with unfinished status in cognition. run

### DISCUSSION THREAD

**Discussion by python2020-maker** [jspsych/jsPsych#1770] [Q&A]
Hi,

I have a problem with unfinished status and the problem is not solved by refreshing. Also worth mentioning that the Timeline is not in green color as what is shown in the Tutorial by Josh, showing unfinished in view.

Is there anybody to help?

**Comment by jodeleeuw** [comment]
Hi @python2020-maker,

We'll need more information here. Can you share some of your code? What do you mean by unfinished status?

(Please use the [discussions board](https://github.com/jspsych/jsPsych/discussions) for Q&A support)

**Reply by python2020-maker** [reply]
I run my experiment on cognition. run, running the experiment was going well but finally the following?


Id | View | Date | Status | Download | Delete
-- | -- | -- | -- | -- | --
#2 | View | 8 hours ago | Unfinished | Download results (.csv) | Delete this run
#1 | View | 8 hours ago | Unfinished | Download results (.csv) | Delete this run

**Comment by amonterou** [comment]
I am encoutering the same problem, even when in the jspsych code I hit the last slide and the test is indeed finished and displayed the data, still the task appears unfinished. Is there a specific command in js that one needs to call so that cognition marks it as finished?

After executing all trials in the timeline I call the displayData
`    jsPsych.init({  
      timeline: device_aware_timeline,
      on_finish: function () {
        jsPsych.data.displayData();
      },
    });`

**Comment by jodeleeuw** [comment]
I don't know how cognition.run handles marking finished vs. unfinished, but you might try removing your `on_finish` function. It's possible that cognition.run needs its own `on_finish` to execute to mark it as complete, and yours is overriding it.

**Reply by amonterou** [reply]
Tried removing the on_finish and ran a test, seems like it is still marking it as unifinished:
![image](https://user-images.githubusercontent.com/56785337/117490760-7b7acf80-af2c-11eb-9c06-777055bce321.png)

**Reply by lily2brain** [reply]
I'm using the `Cognition.run`, and the `on_finish` function is not required, the website itself does that. This must be happening, due to the fact that the participants did not answer all the questions or closed the browser tab during the experiment, in this case, cognition.run appears as not finished. If you download or view individual results that appear as unfinished, you can be more sure of that.

**Reply by amonterou** [reply]
So I was expereimenting and indeed, that is what is happening, I was using the last trial to show the results as a debrief and users just closed the browser window and left the experiment unfinished. I changed my test to use the on_load of that last trial and run this:

`jsPsych.endExperiment(GetResultsScreenString())  `

The GetResultScreenString function does the calculations and returns the html for the results, but this way by the time the user gets the results displayed the experiment is finished and this did the trick.
